python/pyspark/ml/param/__init__.py

```diff
@@ -65,6 +75,146 @@ def __eq__(self, other):
             return False
 
 
+class TypeConverters(object):
+    """
+    .. note:: DeveloperApi
+
+    Factory methods for common type conversion functions for `Param.typeConverter`.
+
+    .. versionadded:: 2.0.0
+    """
+
+    @staticmethod
+    def _is_numeric(value):
+        vtype = type(value)
+        return vtype in [int, float, np.float64, np.int64] or vtype.__name__ == 'long'
+
+    @staticmethod
+    def _is_integer(value):
+        if TypeConverters._is_numeric(value):
+            value = float(value)
+            return value.is_integer()
+        else:
+            return False
+
+    @staticmethod
+    def _can_convert_to_list(value):
+        vtype = type(value)
+        return vtype == list or vtype == np.ndarray or isinstance(value, Vector)
+
+    @staticmethod
+    def _is_string(value):
+        vtype = type(value)
+        return vtype in [str, np.unicode_, np.string_, np.str_] or type(value).__name__ == 'unicode'
+
+    @staticmethod
+    def identity(value):
+        """
+        Dummy converter that just returns value.
+        """
+        return value
+
+    @staticmethod
+    def toList(value):
+        """
+        Convert a value to a list, if possible.
+        """
+        if type(value) == list:
+            return value
+        elif type(value) == np.ndarray:
+            return list(value)
+        elif isinstance(value, Vector):
+            return list(value.toArray())
+        else:
+            raise TypeError("Could not convert %s to list" % value)
+
+    @staticmethod
+    def toListFloat(value):
+        """
+        Convert a value to list of floats, if possible.
+        """
+        if TypeConverters._can_convert_to_list(value):
+            value = TypeConverters.toList(value)
+            if all(map(lambda v: TypeConverters._is_numeric(v), value)):
+                return list(map(lambda v: float(v), value))
```

Comment:
This was necessary because python3 map function returns a "map object" and not a list. But I just changed it to use a list comprehension instead, as it's less confusing.